lib/ddtrace/contrib/sinatra/patcher.rb

```diff
@@ -1,9 +1,40 @@
 # typed: true
 require 'ddtrace/contrib/patcher'
+require 'ddtrace/contrib/sinatra/framework'
+require 'ddtrace/contrib/rack/middlewares'
 
 module Datadog
   module Contrib
     module Sinatra
+      # Set tracer configuration at a late enough time
+      module TracerSetupPatch
+        ONLY_ONCE_PER_APP = Hash.new { |h, key| h[key] = Datadog::Utils::OnlyOnce.new }
```

Comment:
In most scenarios this is probably innocuous, but part of me is concerned about a `Hash` that accumulates `app` references as keys and holds onto them (presumably) indefinitely. Is there any concern for a memory leak?

Is there a better way of doing this? Make the "only_once" an instance variable on the instance perhaps?

Reply:
I think this number is strictly limited by number of Sinatra applications exist. This number will be very small, the vast majority of time 1. I'd be surprised to see this number > 10 across all our users.

Reply:
I concur with @marcotc here. I took this pattern from the Rails integration: see `BEFORE_INITIALIZE_ONLY_ONCE_PER_APP`, `AFTER_INITIALIZE_ONLY_ONCE_PER_APP`.